### PR TITLE
feat: add glob/prefix support to session path mapping

### PR DESCRIPTION
**Pull request by xlyk**
## Summary

- Adds glob pattern support to the `sessions` config field so entries like `~/Code/project.worktrees/*` match multiple directories to the same session name
- Exact paths take priority over globs; among globs, the most specific pattern (longest literal prefix) wins
- Tilde expansion and `path.resolve()` normalization on both keys and cwd
- Empty keys/values silently skipped to prevent ghost matches
- Uses `Bun.Glob` for matching — zero new dependencies
- 23 tests covering exact match, globs (`*`, `**`, `?`, `[...]`, `{a,b}`), specificity ordering, normalization, and edge cases

## Motivation

Users with git worktree setups need to map multiple directories to the same session without adding an explicit config entry per path:

```json
{
  "sessions": {
    "~/Code/project": "my-project",
    "~/Code/project.worktrees/*": "my-project"
  }
}
```

## Test plan

- [x] `bun test src/config.test.ts` — 23/23 passing
- [x] `bunx tsc --noEmit` — clean
- [ ] Manual: add glob entries to `~/.honcho/config.json`, launch Claude Code in matching directories, verify session name via `/honcho:status`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Per-path session name overrides via exact paths and glob patterns for flexible session configuration; exact matches outrank globs and the most specific glob wins; applies only to the per-directory session strategy

* **Documentation**
  * Added "Per-path overrides" guide with example mappings and a pattern-matching reference table

* **Tests**
  * Added comprehensive tests covering exact matches, glob semantics, precedence, normalization, and edge cases

### CODE REVIEW

File: plugins/honcho/src/config.ts [hunk 1]
```diff
@@ -529,10 +543,41 @@ export function getClaudeSettingsDir(): string {
...
+    .sort((a, b) => b.specificity - a.specificity || b.pattern.length - a.pattern.length);
+
+  for (const { pattern, name } of globEntries) {
+    if (new Bun.Glob(pattern).match(normalizedCwd)) {
+      return name;
+    }
+  }
+
+  return null;
+}
```
[the remark this answers is not in the document]
Comment (xlyk) [reply plugins/honcho/src/config.ts:577]: the duplicate normalized key scenario requires someone to manually create conflicting entries (e.g. `~/Code/foo` and `/Users/alice/Code/foo` as separate keys). the behavior is deterministic (first match by JSON insertion order), and switching to Map-based dedup just changes it to last-wins — neither is inherently better. the real fix would be normalizing on write, but that's a larger change. leaving as-is for now.

File: README.md [hunk 2]
```diff
@@ -261,6 +265,27 @@ Session strategy controls how Honcho maps your conversations to sessions. Change
...
+
+| Pattern | Matches |
+| --- | --- |
+| `/Users/alice/Code/foo` | Exact path only |
+| `~/Code/foo` | Same (tilde expands to home dir) |
+| `~/Code/*` | Any direct child of `~/Code/` |
+| `~/Code/**` | Any descendant at any depth under `~/Code/` |
+| `~/Code/work-*` | Directories starting with `work-` in `~/Code/` |
+
+Exact matches always take priority over globs. Among globs, the most specific pattern (longest literal prefix before the first wildcard) wins. Per-path overrides only apply to the `per-directory` strategy.
```
[the remark this answers is not in the document]
Comment (xlyk) [reply README.md:287]: addressed in 2514529 — relative paths are now rejected instead of silently anchored. only keys starting with `/` or `~` are considered, so there's nothing to document.

File: plugins/honcho/src/config.ts [hunk 3]
```diff
@@ -8,6 +8,20 @@ function sanitizeForSessionName(s: string): string {
...
 }
 
+const GLOB_META_RE = /[*?[\]{]/;
+
+function normalizePath(p: string): string {
+  let result = p;
+  if (result.startsWith("~/") || result === "~") {
+    result = join(homedir(), result.slice(1));
+  }
+  return resolve(result);
```
[the remark this answers is not in the document]
Comment (xlyk) [reply plugins/honcho/src/config.ts:18]: good catch — fixed in 9db610a. filter now uses `key === "~" || key.startsWith("~/") || isAbsolute(key)` to match exactly what `normalizePath` supports.
